flink-runtime/src/main/java/org/apache/flink/runtime/util/ExecutorThreadFactory.java

```diff
@@ -106,6 +109,8 @@ public ExecutorThreadFactory(String poolName, UncaughtExceptionHandler exception
 
 	@Override
 	public Thread newThread(Runnable runnable) {
+		runnable = new RunnableMDCWrapper(runnable);
+
```

Comment:
Will this work correctly if the `ExecutorThreadFactory` is used in some thread pool that is shared between different jobs? 

I think at best, we would loose the `MDC`. At worst, if new pooled worker is spawned when in the scope of Job1's MDC, this MDC would stick with this worker forever, regardless if it's executing code belonging to some another job?

Reply:
I think so, because the wrapper resets the MDC after it has finished.

Reply:
That said, we shouldn't just override the MDC of the thread; if the owner of the pool (e.g., TM) has something in the MDC then it should be preserved. (i.e., merge the MDCs instead)